File: alphatimer-ares-serial-api/src/main/java/org/lisasp/alphatimer/api/ares/serial/events/dropped/DataHandlingMessage1DroppedEvent.java
package org.lisasp.alphatimer.api.ares.serial.events.dropped;

import lombok.RequiredArgsConstructor;
import lombok.Value;
import org.lisasp.alphatimer.api.ares.serial.data.Bytes;

import java.time.LocalDateTime;

@Value
@RequiredArgsConstructor
public class DataHandlingMessage1DroppedEvent implements InputDataDroppedEvent {
    private final LocalDateTime timestamp;
    private final String competition;
    private final String message;
    private final Bytes data;

    public DataHandlingMessage1DroppedEvent(LocalDateTime timestamp, String competition, String message, byte[] data) {
        this(timestamp, competition, message, new Bytes(data));
    }
}

File: alphatimer-protocol-api/src/main/java/org/lisasp/alphatimer/api/protocol/events/dropped/DataHandlingMessage1DroppedEvent.kt
package org.lisasp.alphatimer.api.protocol.events.dropped

import org.lisasp.alphatimer.api.protocol.data.Bytes

data class DataHandlingMessage1DroppedEvent(private val message: String, private val data: Bytes) : InputDataDroppedEvent {
    constructor(message: String, data: ByteArray) : this(message, Bytes(data))
}
